zerver/lib/actions.py

```diff
@@ -740,22 +740,32 @@ def do_set_realm_authentication_methods(realm: Realm,
 def do_set_realm_message_editing(realm: Realm,
                                  allow_message_editing: bool,
                                  message_content_edit_limit_seconds: int,
-                                 allow_community_topic_editing: bool) -> None:
+                                 allow_community_topic_editing: bool,
+                                 acting_user: Optional[UserProfile]=None) -> None:
     realm.allow_message_editing = allow_message_editing
     realm.message_content_edit_limit_seconds = message_content_edit_limit_seconds
     realm.allow_community_topic_editing = allow_community_topic_editing
-    realm.save(update_fields=['allow_message_editing',
-                              'allow_community_topic_editing',
-                              'message_content_edit_limit_seconds',
-                              ],
-               )
+
+    event_time = timezone_now()
+    updated_properties = dict(allow_message_editing=allow_message_editing,
+                              message_content_edit_limit_seconds=message_content_edit_limit_seconds,
+                              allow_community_topic_editing=allow_community_topic_editing)
+
+    for updated_property, updated_value in updated_properties.items():
+        old_value = getattr(realm, updated_property)
```

Comment:
I'm pretty sure the `old_value` here is fetched too late.

I think this logic is also going to end up logging "changes" even when a field's value wasn't changed in this request.

Reply:
Usually, before calling these functions we check that the previous value should not be the same as the updated value still if want we can do a check here and do an early return.